src/main/java/dev/plex/extras/command/ClownfishCommand.java

```diff
@@ -0,0 +1,121 @@
+    {
+        if (args.length == 0)
+        {
+            ItemStack clownfish = new ItemStack(Material.TROPICAL_FISH);
+            ItemMeta meta = clownfish.getItemMeta();
+
+            meta.displayName(Component.text("Clownfish"));
+            clownfish.setItemMeta(meta);
+
+            player.getInventory().addItem(clownfish);
+            return MiniMessage.miniMessage().deserialize("<rainbow>blub blub... ><_>");
+        }
+        else if (args[0].equals("toggle"))
+        {
+            List<String> toggledPlayers = TFMExtras.getModule().getConfig().getStringList("server.clownfish.toggled_players");
+
+            boolean isToggled = toggledPlayers.contains(player.getUniqueId().toString());
+            if (isToggled)
+            {
+                toggledPlayers.remove(player.getUniqueId().toString());
+            }
+            else
+            {
+                toggledPlayers.add(player.getUniqueId().toString());
+            }
+
+            TFMExtras.getModule().getConfig().set("server.clownfish.toggled_players", toggledPlayers);
+            TFMExtras.getModule().getConfig().save();
+
+            return messageComponent("toggleClownfish", isToggled ? "now" : "no longer");
+        }
+        else if (args[0].equals("restrict") && args.length == 2)
+        {
+            if (silentCheckPermission(commandSender, "plex.tfmextras.restrictclownfish"))
+            {
+                PlexPlayer target = DataUtils.getPlayer(args[1]);
+                if (target == null)
+                {
+                    throw new PlayerNotFoundException();
+                }
+
+                List<String> restrictedPlayers = TFMExtras.getModule().getConfig().getStringList("server.clownfish.restricted");
+
+                boolean isRestricted = restrictedPlayers.contains(target.getUuid().toString());
+                if (isRestricted)
+                {
+                    restrictedPlayers.remove(target.getUuid().toString());
+                }
+                else
+                {
+                    restrictedPlayers.add(target.getUuid().toString());
+                }
+
+                TFMExtras.getModule().getConfig().set("server.clownfish.restricted", restrictedPlayers);
+                TFMExtras.getModule().getConfig().save();
+
+                return messageComponent("restrictClownfish", target.getName(), isRestricted ? "now" : "no longer");
+            }
+            else
+            {
+                return MiniMessage.miniMessage().deserialize("<red>You do not have permission to use this command.");
+            }
+        }
+        else
+        {
+            return usage();
+        }
+    }
+
+    @Override
+    public @NotNull List<String> smartTabComplete(@NotNull CommandSender sender, @NotNull String alias, @NotNull String[] args) throws IllegalArgumentException {
+        if (silentCheckPermission(sender, "plex.tfmextras.restrictclownfish"))
```

Comment:
If anyone can use `/clownfish toggle`, this should be merged with the if-statement on L108, so players can't get the list of names when running `/clownfish restrict`.

Reply:
This will also make L114-L117 redundant.

Reply:
As per discussion with @notsceptor, I made a mistake with this review so dismissing this.